reservation/views.py

```diff
@@ -1,28 +1,100 @@
-from django.contrib.auth.decorators import login_required
-from django.shortcuts import render, get_object_or_404
-from django.utils import timezone
+from datetime import datetime
 
-from cinema.models import Movie, Schedule
-from reservation.forms import ReservationForm
+from django.core.mail import EmailMessage
+from django.template.loader import get_template
+from django.template import Context
+from django.shortcuts import render, get_object_or_404
+from cinema.models import Movie, Schedule, Seat
+from reservation.forms import ReservationFilterForm
+from django.contrib import messages
+from reservation.models import Reservation
 
 
-@login_required
 def reservation_options(request, pk):
     movie = get_object_or_404(Movie, pk=pk)
-    schedule_list = Schedule.objects.filter(
-        movie=movie,
-        playing_time__gt=timezone.now()
-    ).order_by('playing_time')
-    form = ReservationForm()
-    if form.is_valid():
-        form.save(commit=False)
-        form.user_name = request.user
-        form.save()
+    form = ReservationFilterForm(movie_id=movie)
+    schedule_list = None
 
+    if request.method == 'POST':
+        cinema = request.POST.get('cinema')
+        schedule_list = Schedule.objects.filter(
+            movie=movie,
+            cinema=cinema,
+            playing_time__gte=datetime.now()).order_by('playing_time')
     context = {
         'movie': movie,
         'schedule': schedule_list,
         'form': form,
     }
 
     return render(request, 'reservation/reservation_options.html', context)
+
+
+def select_seats(request, movie, schedule):
+    movie_object = get_object_or_404(Movie, pk=movie)
+    schedule_object = get_object_or_404(Schedule, pk=schedule)
+
+    hall = Schedule.objects.values_list('hall', flat=True).get(id=schedule)
+    seat_list = Seat.objects.filter(hall=hall)
+
+    # Get the already booked seats, so we can block them in the template
+    existent_reservations = Reservation.objects.filter(details=schedule_object)
+    occupied_seat_ids = []
+    for item in existent_reservations:
+        occupied_seat_ids.append(item.seat_id)
```

Comment:
This list could also be achieved with the help of values_list, flat.

Reply:
Yep, got it now. Thanks!